### ISSUE TITLE
Communication issue with BLE device

### ISSUE THREAD

**Issue by BrainDamage** [nullp0tr/bluew#7]
Hello, I'm trying to communicate with a BLE device, the code is pretty simple, essentially a straight copy of the example, unfortunately, it rises an exception:

```
bluew.write_attribute("FE:8D:5D:C5:CB:C3","0000fff6-0000-1000-8000-00805f9b34fb",[0x01,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x01])

/usr/lib/python3.6/site-packages/bluew/api.py in write_attribute(mac, attribute, data, *args, **kwargs)
    109 
    110     with Connection(mac, *args, **kwargs) as connection:
--> 111         return connection.write_attribute(attribute, data)
    112 
    113 

/usr/lib/python3.6/site-packages/bluew/connections.py in wrapper(self, *args, **kwargs)
     33         except Exception as exp:
     34             self.close()
---> 35             raise exp
     36     return wrapper
     37 

/usr/lib/python3.6/site-packages/bluew/connections.py in wrapper(self, *args, **kwargs)
     30         """This function wraps the function passed to close_on_error()"""
     31         try:
---> 32             return func(self, *args, **kwargs)
     33         except Exception as exp:
     34             self.close()

/usr/lib/python3.6/site-packages/bluew/connections.py in write_attribute(self, attribute, data)
     84     def write_attribute(self, attribute, data):
     85         """Write to a bluetooth attribute."""
---> 86         return self.engine.write_attribute(self.mac, attribute, data)
     87 
     88     @close_on_error

/usr/lib/python3.6/site-packages/bluew/dbusted/decorators.py in _wrapper(self, dev, *args, **kwargs)
     22         if '/dev_' not in dev:
     23             dev = '/dev_' + dev.replace(':', '_')
---> 24         return func(self, dev, *args, **kwargs)
     25     return _wrapper
     26 

/usr/lib/python3.6/site-packages/bluew/dbusted/decorators.py in _wrapper(self, dev, *args, **kwargs)
     33         available = self._is_device_available(dev)
     34         if available:
---> 35             return func(self, dev, *args, **kwargs)
     36         raise DeviceNotAvailable(name=self.name, version=self.version)
     37     return _wrapper

/usr/lib/python3.6/site-packages/bluew/dbusted/decorators.py in _wrapper(self, *args, **kwargs)
     67     def _wrapper(self, *args, **kwargs):
     68         try:
---> 69             return func(self, *args, **kwargs)
     70         except IfaceError as exp:
     71             # pylint: disable=W0212

/usr/lib/python3.6/site-packages/bluew/dbusted/dbusted.py in write_attribute(self, mac, attribute, data)
    374         """
    375 
--> 376         path = self._uuid_to_path(attribute, mac)
    377         gattchrciface = BluezGattCharInterface(self._bus, path)
    378         gattchrciface.write_value(data)

/usr/lib/python3.6/site-packages/bluew/dbusted/dbusted.py in _uuid_to_path(self, uuid, dev)
    517 
    518     def _uuid_to_path(self, uuid, dev):
--> 519         path = self._timeout(self._get_attr_path, self.timeout)(uuid, dev)
    520         if not path:
    521             raise DeviceNotAvailable(self.name, self.version)

/usr/lib/python3.6/site-packages/bluew/dbusted/dbusted.py in _wrapper(*args, **kwargs)
    539             current_time = time.time()
    540             while (current_time < start_time + timeout) and not result:
--> 541                 result = func(*args, **kwargs)
    542                 current_time = time.time()
    543             return result

/usr/lib/python3.6/site-packages/bluew/dbusted/dbusted.py in _get_attr_path(self, uuid, dev)
    508 
    509     def _get_attr_path(self, uuid, dev):
--> 510         chrcs = self.get_chrcs(dev)
    511         try:
    512             chrc = list(filter(lambda chrc_: uuid == chrc_.UUID, chrcs))[0]

/usr/lib/python3.6/site-packages/bluew/dbusted/decorators.py in _wrapper(self, dev, *args, **kwargs)
     22         if '/dev_' not in dev:
     23             dev = '/dev_' + dev.replace(':', '_')
---> 24         return func(self, dev, *args, **kwargs)
     25     return _wrapper
     26 

/usr/lib/python3.6/site-packages/bluew/dbusted/dbusted.py in get_chrcs(self, mac)
    291 
    292         boiface = BluezObjectInterface(self._bus)
--> 293         return boiface.get_characteristics(mac)
    294 
    295     @mac_to_dev

/usr/lib/python3.6/site-packages/bluew/dbusted/interfaces.py in get_characteristics(self, dev)
    628         objects = list(filter(lambda x: dev in x.get('Path', None), objects))
    629         characteristics = tuple(
--> 630             map(lambda obj: BLECharacteristic(**obj), objects))
    631         return characteristics
    632 

/usr/lib/python3.6/site-packages/bluew/dbusted/interfaces.py in <lambda>(obj)
    628         objects = list(filter(lambda x: dev in x.get('Path', None), objects))
    629         characteristics = tuple(
--> 630             map(lambda obj: BLECharacteristic(**obj), objects))
    631         return characteristics
    632 

/usr/lib/python3.6/site-packages/bluew/characteristics.py in __init__(self, **kwargs)
     21         attrs = {'Value', 'Flags', 'Notifying',
     22                  'Service', 'UUID', 'Path', 'NotifyAcquired'}
---> 23         super().__init__(attrs, **kwargs)
     24         self.value = kwargs.get('Value')
     25         self.flags = kwargs.get('Flags')

/usr/lib/python3.6/site-packages/bluew/ppobj.py in __init__(self, attrs, **kwargs)
     20                 name = self.__class__.__name__
     21                 raise TypeError(
---> 22                     '{a} should not have attribute {b}'.format(a=name, b=key))
     23             setattr(self, key, value)
     24 

TypeError: BLECharacteristic should not have attribute WriteAcquired
```

I do know that both the mac and gatt service is valid, confirmed in bluetoothctl:

```
[B031_02 C5CBC3]# connect FE:8D:5D:C5:CB:C3
Attempting to connect to FE:8D:5D:C5:CB:C3
Connection successful
[B031_02 C5CBC3]# list-attributes
Primary Service
        /org/bluez/hci0/dev_FE_8D_5D_C5_CB_C3/service0009
        0000fff0-0000-1000-8000-00805f9b34fb
        Unknown
Characteristic
        /org/bluez/hci0/dev_FE_8D_5D_C5_CB_C3/service0009/char000d
        0000fff6-0000-1000-8000-00805f9b34fb
        Unknown
Characteristic
        /org/bluez/hci0/dev_FE_8D_5D_C5_CB_C3/service0009/char000a
        0000fff7-0000-1000-8000-00805f9b34fb
        Unknown
Descriptor
        /org/bluez/hci0/dev_FE_8D_5D_C5_CB_C3/service0009/char000a/desc000c
        00002902-0000-1000-8000-00805f9b34fb
        Client Characteristic Configuration
Primary Service
        /org/bluez/hci0/dev_FE_8D_5D_C5_CB_C3/service0008
        00001801-0000-1000-8000-00805f9b34fb
        Generic Attribute Profile

[B031_02 C5CBC3]# select-attribute 0000fff6-0000-1000-8000-00805f9b34fb

[B031_02 C5CBC3:/service0009/char000d]# write 0x01 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x01
Attempting to write fd 7
```

since the device answers to the command, I assume that bluetoothctl's write is at least working, what am I doing wrong?
Is the device somehow misreporting some write attribute flags?

Python 3.6.4, bluew 0.4.3

**Comment by nullp0tr** [comment]
Hey, you're doing noting wrong, I'll upload a patch to fix this in a bit, it's basically just a mechanism to notify me about new bluez flags that aren't included but I'll take out.

**Comment by nullp0tr** [comment]
This should be fixed by version-0.4.4. Can you please `pip install --upgrade bluew` and let me know if the error still persists?

**Comment by BrainDamage** [comment]
It's indeed fixed with 0.4.4, thanks

**Comment by nullp0tr** [comment]
Glad to hear:) Let me know if you have any more issues.
